lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -878,4 +881,57 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+    
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     * 
+     * @param  string $query
+     * 
+     * @return string
+     */
+    private function getOrderByForOver($query)
+    {
+        //preset empty array
+        $orderByArray = [];
+
+        // remove extra white space
+        $query = preg_replace('/\s+/', ' ', $query);
+
+        //determine if 'ORDER BY' is part of the query
+        $orderByFound = preg_match('/(\bORDER\b)(\s*)(\bBY\b)(?!.*\b\1\b)/i', $query, $orderByPositionArray, PREG_OFFSET_CAPTURE);
```

Comment:
I believe it's better to use `preg_match_all()` and use the last match. Otherwise, a query like `ORDER BY my_order` will be probably ignored.

Reply:
Aren't `\b`'s around `\s` redundant (`\b)(\s*)(\b`)? Could it be just `\s+`?

Reply:
Just tying to address possible whitespaces between ORDER and BY

Reply:
You're right, `my_order` will not match `\bORDER\b`.

Reply:
This one can stay? Just checking!